occt-import-js/src/js-interface.cpp

```diff
@@ -73,19 +73,19 @@ class HierarchyWriter
                     indexArr.set (triangleCount * 3 + 2, vertexOffset + v2);
                     triangleCount += 1;
                 });
+                emscripten::val faceColorObj (emscripten::val::object ());
+                faceColorObj.set ("first", triangleOffset);
+                faceColorObj.set ("last", triangleCount - 1);
                 Color faceColor;
                 if (face.GetColor (faceColor)) {
-                    emscripten::val faceColorObj (emscripten::val::object ());
-                    faceColorObj.set ("first", triangleOffset);
-                    faceColorObj.set ("last", triangleCount - 1);
                     emscripten::val colorArr (emscripten::val::array ());
                     colorArr.set (0, faceColor.r);
                     colorArr.set (1, faceColor.g);
                     colorArr.set (2, faceColor.b);
                     faceColorObj.set ("color", colorArr);
-                    faceColorArr.set (faceColorCount, faceColorObj);
-                    faceColorCount += 1;
                 }
+                faceColorArr.set (faceColorCount, faceColorObj);
```

Comment:
Now that it contains all the faces (even the ones without color) it's a bit misleading to call this array `face_colors`. I would rename the variable here to `brepFacesArr` and the key in the resulting json to `brep_faces`.

Reply:
Would simply `faces` also be fine or do you specifically want `breb_faces`?

Reply:
I prefer `brep_faces` to make it clear that it's not about the tesselated triangles, but the original faces.